src/QsCompiler/CompilationManager/CompilationUnitManager.cs

```diff
@@ -141,14 +141,34 @@ public void Dispose()
         // routines related to tracking the source files
 
         /// <summary>
-        /// Returns the string with the file ID associated with the given URI used throughout the compilation.
+        /// Converts a URI into the file ID used during compilation if the URI is an absolute file URI. 
         /// </summary>
-        public static bool TryGetFileId(Uri uri, out NonNullable<string> id)
+        /// <exception cref="ArgumentNullException">Thrown if the URI is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the URI is not an absolute URI or not a file URI.</exception>
+        public static NonNullable<string> GetFileId(Uri uri) =>
+            uri is null
+                ? throw new ArgumentNullException(nameof(uri))
+                : TryGetFileId(uri, out var fileId)
+                ? fileId
+                : throw new ArgumentException("The URI is not an absolute file URI.", nameof(uri));
+
+        /// <summary>
+        /// Converts a URI into the file ID used during compilation if the URI is an absolute file URI. 
+        /// </summary>
+        /// <returns>True if converting the URI to a file ID succeeded.</returns>
+        [Obsolete("Use GetFileId instead after ensuring that the URI is an absolute file URI.")]
```

Comment:
What is the reason for the API change?

Reply:
[See here.](https://github.com/microsoft/qsharp-compiler/pull/468#issuecomment-647790459)

Reply:
ok, thanks.